src/app/services/tvdb.service.ts

```diff
@@ -1,22 +1,25 @@
 import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
 import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/do';
 import { TvShowData } from '../tvshows/store/tvshows.models';
 
-const TVSHOWS: TvShowData[] = [
-  { title: 'WalkingDead' },
-  { title: 'Wheel of Forune' },
-  { title: 'The Walking Dead' },
-  { title: 'Rick & Morty' },
-];
+const API_BASE_URL = 'http://api.tvmaze.com';
 
 @Injectable()
 export class TvDbService {
+
+  constructor(private http: Http) { }
+
   search(query: string): Promise<TvShowData[]> {
-    const regexp = new RegExp(query, 'i');
-    console.log(regexp);
-    const results = [...TVSHOWS].filter((show) => regexp.test(show.title));
-    return new Promise((resolve, reject) => {
-      setTimeout(() => resolve(results), 1000); // Mock 3sec API request
-    });
+    const q = encodeURI(query);
+    return this.http
+      .get(`${API_BASE_URL}/search/shows?q=${q}`)
+      .map((response) => response.json())
+      .map((shows) => shows.map((item) => ({
+        title: item.show.name,
+      })))
+      .toPromise();
```

Comment:
I'd suggest keeping this as an observable. The usage of promise in Angular 2+ would be an anti-pattern (imo).